slsim/Sources/source.py

```diff
@@ -125,6 +137,80 @@ def kwargs_variability_extracted(self):
                             "MJD": times,
                             name: magnitudes,
                         }
+
+            # Check if AGN model is used
+            elif any(
+                element in kwargs_variability_list_agn
+                for element in list(self.kwargs_variability)
+            ):
+
+                z = self.source_dict["z"]
+                if self.cosmo is None:
+                    raise ValueError(
+                        "Cosmology cannot be None for AGN class. Please"
+                        "provide a suitable astropy cosmology."
+                    )
+
+                else:
+                    # Get the dict object from the astropy.table.column
+                    agn_kwarg_dict = self.source_dict["kwargs_agn_model"].data[0][0]
```

Comment:
You can demand kwargs_agn_model keywords from the kwargs_variability (this is where user should put all the keywords for variability model) and based on those keywords construct agn_kwargs_dict using source_dict where the values of these keyswords should be available. In source_dict, these values should be placed in different column instead of a compact dictionary.

Reply:
Got it! 

I wrote a method at the bottom of source.py to contain the extraction of AGN parameters.